### PR TITLE
Update Python versions in CI

### PR DESCRIPTION
**Pull request by dwhswenson**
Aside from being overdue and needed, this is largely in response to a CI failure due to a change in mdtraj-dev. If this is just that our Python CI is out of date, then the problem is on us. If we're still failing after this PR, then we'll take it upstream.

### CODE REVIEW

**Comment by dwhswenson** [comment]
Sent upstream in https://github.com/mdtraj/mdtraj/pull/1913.

**Comment by dwhswenson** [comment]
Been having more failures lately my guess is incompatibility of new `ubuntu-latest` with some old Python versions. This seems to fit, and brings makes us compliant with SPEC0.

This is ready for review and comment. Without review, I will merge this after at least 24 hours, no sooner than Sat 21 Dec 15:00 GMT (09:00 my local).

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -23,22 +23,22 @@ jobs:
...
             OS: "ubuntu"
             MDTRAJ: "mdtraj-dev"
             INSTALL_WITH: "pip"
-          - CONDA_PY: 3.9
+          - CONDA_PY: "3.12"
             OS: "windows"
             MDTRAJ: "mdtraj-release"
             INSTALL_WITH: "conda"
-          - CONDA_PY: 3.7
+          - CONDA_PY: "3.10"
```
Comment (sroet) [inline .github/workflows/unit-tests.yml:41]: Any reason to test outside of the spec0 matrix?

**Review by sroet — APPROVED** [review]
1 question, LGTM otherwise

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (sroet) [reply .github/workflows/unit-tests.yml:41]:
```suggestion
          - CONDA_PY: "3.11"
```

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -23,22 +23,22 @@ jobs:
...
         MDTRAJ: ["mdtraj-release"]
         INSTALL_WITH: ["pip"]
         include:
-          - CONDA_PY: 3.9
+          - CONDA_PY: "3.12"
             OS: "ubuntu"
             MDTRAJ: "mdtraj-dev"
             INSTALL_WITH: "pip"
-          - CONDA_PY: 3.9
+          - CONDA_PY: "3.12"
```
Comment (dwhswenson) [inline .github/workflows/unit-tests.yml:37]:
```suggestion
          - CONDA_PY: "3.13"
```

**Pull request merged**